test/test.util.js

```diff
@@ -71,10 +77,11 @@ function testListIsPunycode (t, list) {
 }
 
 function testListDoesntContainRepeats (t, list) {
-  list.forEach((domain) => {
-    const count = list.filter(item => item === domain).length
-    t.ok(count === 1, `domain '${domain}' is duplicated. Domains can only appear in list once`)
-  })
+  const clone = [...list];
+  clone.sort();
+  for (let i = 0; i < clone.length - 1; i++) {
+    if (clone[i] === clone[i + 1]) t.fail(`domain ${clone[i]} is duplicated. Domains can only appear in the list once`)
+  }
```

Comment:
nb. might a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) or plain object be faster, ie. O(n) vs. O(n log n)?

Reply:
```suggestion
  const clone = new Set(list);

  if (clone.size === list.length) {
    return;
  }

  for (const item of list) {
    if (clone.has(item)) {
      clone.delete(item);
    } else {
      t.fail(`domain ${item} is duplicated. Domains can only appear in the list once`);
    }
  }
```